### PR TITLE
Unit tests for `getTopHeadlines` controller with mocked API calls

### PR DESCRIPTION
**Pull request by bojana-petroska**
This pull request adds unit tests for the `getTopHeadlines` controller in the News module, mocking the `fetchFromNewsAPI` function and testing various response scenarios.

- Mocked the fetchFromNewsAPI function to simulate API calls.

**Added tests for:**
- Successful Fetch: Verifies a `200` response with articles.
- No Articles Found: Ensures an empty array is returned with a `200` status.
- Error Handling: Simulates an error and checks for a `500` status.
- Query Parameter Validation: Tests invalid query parameters, ensuring a `400` status.

Screenshot 2024-12-07 at 14 39 27

P.S. The only changed file is `src/tests/news/getTopHeadlinesController.test.ts`. The other files are coming from the previous PR. Please only review this file.

### CODE REVIEW

File: src/tests/news/getTopHeadlinesController.test.ts
Comment (shenoac) [inline src/tests/news/getTopHeadlinesController.test.ts:1]: I ran the tests and they passed :)

**Review by regikov — COMMENTED** [review]
looks great to me! I would probably have added test for missing query parameters too.

File: src/tests/news/getTopHeadlinesController.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import app from '../../index.js';
+import request from 'supertest';
+import { Request, Response } from 'express';
+import { fetchFromNewsAPI } from '../../utils.js';
+import newsController from '../../modules/News/controller.js';
+
+jest.mock('../../utils.ts');
+jest.mock('../../middleware/auth.ts', () =>
```
Comment (shenoac) [inline src/tests/news/getTopHeadlinesController.test.ts:8]: Nice use of the mocking!

**Comment by bojana-petroska** [comment]
thanks @shenoac !
thanks @regikov ! in the validation the parameters are optional and therefore the tests would not work. 
@ahzia maybe we can see this together, how to test optional parameters?

File: src/tests/news/getTopHeadlinesController.test.ts
[the remark this answers is not in the document]
Comment (bojana-petroska) [reply src/tests/news/getTopHeadlinesController.test.ts:1]: yes, a naming convention would be nice, it seemed very lengthly to me too.

[the remark this answers is not in the document]
Comment (ahzia) [reply src/tests/news/getTopHeadlinesController.test.ts:1]: @bojana-petroska topHeadlines.test.ts would be fine

[the remark this answers is not in the document]
Comment (ahzia) [reply src/tests/news/getTopHeadlinesController.test.ts:1]: And If it is a mock test then: `topHeadlines-mock.test.ts`

File: src/modules/News/controller.ts [hunk 2]
```diff
@@ -13,6 +17,35 @@ const getLatestNews = async (req: Request, res: Response) => {
...
+      currentPage: paginatedArticles.currentPage,
+      totalPages: paginatedArticles.totalPages,
+    });
+  } catch (error) {
+    res
+      .status(500)
+      .send({ message: 'Error in getting personalized news', error });
+  }
+};
+
```
Comment (ahzia) [inline src/modules/News/controller.ts:48]: Comment 1: The commit that contains these parts of the code shouldn't be in this PR, as they are related to the other [PR ](https://github.com/StartSteps-Digital-Education-GmbH/News-App/pull/14) you are working on.
When working on a new feature always create the new branches from the `main` branch.
For now, it's fine but we have to merge your personalized news branch first, then this one.

File: src/modules/News/routes.ts
Comment (ahzia) [inline src/modules/News/routes.ts:1]: same as Comment 1

File: src/modules/News/validation.ts
Comment (ahzia) [inline src/modules/News/validation.ts:1]: same as Comment 1

File: src/utils.ts
Comment (ahzia) [inline src/utils.ts:1]: Same as Comment 1

**Review by ahzia — APPROVED** [review]
Good job @bojana-petroska 👏
The mock tests are perfectly written, just rename the file to `topHeadlines-mock.test.ts`
Take a look at my suggestions in inline comments and consider it on your next PR.

**Comment by ahzia** [comment]
@bojana-petroska you need to do a sync on this branch before I merge.

**Review by ahzia — APPROVED** [review]
Good Job @bojana-petroska I am merging your branch!

**Pull request merged**